pkg/deployer/lib/extension/operation.go

```diff
@@ -0,0 +1,44 @@
+// SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Gardener contributors.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package extension
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	lsv1alpha1 "github.com/gardener/landscaper/apis/core/v1alpha1"
+)
+
+type Operation struct {
```

Comment:
not sure if we should just rather add all the values as function parameters.
Also I think we should only call the hook with the context parameters like the context, the deployitem and target.
other values should be available within the hook implementation itself